src/pim-types.ts

```diff
@@ -255,6 +264,57 @@ export class TaskType extends EventType {
     ret.color = this.color;
     return ret;
   }
+
+  public getNextOccurence(): TaskType | null {
+    if (!this.isRecurring()) {
+      return null;
+    }
+
+    const rrule = this.rrule.clone();
+
+    if (rrule.count && rrule.count <= 1) {
+      return null; // end of reccurence
+    }
+
+    rrule.count = null; // clear count so we can iterate as many times as needed
+    const recur = rrule.iterator(this.startDate ?? this.dueDate);
+    let next = recur.next();
+    while ((next = recur.next())) {
+      if (next.compare(ICAL.Time.now()) > 0) {
+        break;
+      }
+    }
+
+    if (!next) {
+      return null; // end of reccurence
+    }
+
+    const nextStartDate = this.startDate && next;
+    const nextDueDate = this.dueDate && next;
+    if (nextStartDate && nextDueDate) {
+      const offset = (this.dueDate as ICAL.Time).subtractDateTz(this.startDate);
```

Comment:
Why this cast?

Reply:
`this.dueDate` can be `undefined`. This branch will only run if `nextDueDate` is defined (which means `this.dueDate` is defined), but Typescript doesn't know that so it complains.

Reply:
As I said elsewhere, the correct way of doing this in typescript is the `!` operator.
Also, check to see if typescript is smart enough to understand it is indeed not null once you change to the trenary operator as you suggested. If it is, great, if not, do the above.